php/php.dbgp/src/org/netbeans/modules/php/dbgp/breakpoints/BreakpointModel.java

```diff
@@ -179,21 +180,56 @@ private boolean foundBreakpoint(DebugSession session, Acceptor acceptor) {
                 continue;
             }
             if (acceptor.accept(breakpoint)) {
-                AbstractBreakpoint abpnt = (AbstractBreakpoint) breakpoint;
-                synchronized (myCurrentBreakpoints) {
-                    AbstractBreakpoint bpnt = myCurrentBreakpoints.get(session);
-                    myCurrentBreakpoints.put(session, abpnt);
-                    fireChangeEvents(new ModelEvent[]{
-                        new ModelEvent.NodeChanged(this, bpnt),
-                        new ModelEvent.NodeChanged(this, abpnt)
-                    });
-                }
+                updateCurrentBreakpoint(session, breakpoint);
                 return true;
             }
         }
         return false;
     }
 
+    public void setCurrentBreakpoint(DebugSession session, String id) {
+        Breakpoint[] breakpoints = DebuggerManager.getDebuggerManager().getBreakpoints();
+        for (Breakpoint breakpoint : breakpoints) {
+            if (!(breakpoint instanceof AbstractBreakpoint)) {
+                continue;
+            }
+            if (breakpoint.getValidity().equals(VALIDITY.INVALID)) {
+                continue;
+            }
+            if (!((AbstractBreakpoint) breakpoint).isSessionRelated(session)) {
+                continue;
+            }
+            if (!((AbstractBreakpoint) breakpoint).isEnabled()) {
+                continue;
+            }
+            if (((AbstractBreakpoint) breakpoint).getBreakpointId().equals(id)) {
+                updateCurrentBreakpoint(session, breakpoint);
+            }
+        }
+    }
+
+    private void updateCurrentBreakpoint(DebugSession session, Breakpoint breakpoint) {
+        AbstractBreakpoint abpnt = (AbstractBreakpoint) breakpoint;
+        synchronized (myCurrentBreakpoints) {
+            AbstractBreakpoint bpnt = myCurrentBreakpoints.get(session);
+            myCurrentBreakpoints.put(session, abpnt);
+            fireChangeEvents(new ModelEvent[]{
+                new ModelEvent.NodeChanged(this, bpnt),
+                new ModelEvent.NodeChanged(this, abpnt)
+            });
+        }
+    }
+
+    public void setSearchCurrentBreakpointById(Boolean flag) {
+        synchronized (this) {
+            searchCurrentBreakpointById = flag;
+        }
+    }
+
+    public Boolean isSearchCurrentBreakpointById() {
```

Comment:
```suggestion
    @CheckForNull
    public Boolean isSearchCurrentBreakpointById() {
```

Reply:
BTW, do we need to allow `null` for this field? I mean, cannot it be simply a `boolean`?

Reply:
If I get it right, I would simply define it as `boolean` with default value `false`; and, if this feature is enabled/supported, it would be set to `true`. So, no need for `null`.

Reply:
I agree with Tomas. If we use `Boolean`, we have to check `null` every time. It seems that `null` is not set. So, we can use `boolean`, I think (if I didn't overlook anything).